subsys/bluetooth/mesh/shell/shell_utils.c

```diff
@@ -48,6 +51,74 @@ bool shell_model_str2bool(const char *str)
 	return str2u8(str);
 }
 
+
+int shell_model_str2sensorval(const char *str, struct sensor_value *out)
+{
+	int32_t sign = 1;
+	char buf[18]; /* 10 digits + decimal point + 6 digits + terminator */
+
+	if (str != NULL && *str == '-') {
```

Comment:
`strlen` that is called later doesn't expect string to be NULL, therefore checking `str` for NULL doesn't give much sense.